src/tribler/core/upgrade/upgrade.py

```diff
@@ -68,6 +70,22 @@ def cleanup_noncompliant_channel_torrents(state_dir):
                                  file_path)
 
 
+def catch_db_is_corrupted_exception(upgrader_method):
```

Comment:
Could you please add a description for this decorator with the notion that it could only apply to TriblerUpgradfer methods?

Reply:
I added a comment to it:

```python
    # This decorator applied for TriblerUpgrader methods. It suppresses and remembers the DatabaseIsCorrupted exception.
    # As a result, if one upgrade method raises an exception, the following upgrade methods are still executed.
    #
    # The reason for this is the following: it is possible that one upgrade method upgrades database A
    # while the following upgrade method upgrades database B. If a corruption is detected in the database A,
    # the database B still needs to be upgraded. So, we want to temporarily suppress the DatabaseIsCorrupted exception
    # until all upgrades are executed.
    #
    # If an upgrade finds the database to be corrupted, the database is marked as corrupted. Then, the next upgrade
    # will rename the corrupted database file (the get_db_version call handles this) and immediately return because
    # there is no database to upgrade. So, if one upgrade function detects database corruption, all the following
    # upgrade functions for this specific database will skip the actual upgrade. As a result, a new database with
    # the current DB version will be created on the Tribler Core start.
```